src/Router/WithMediator.tsx

```diff
@@ -0,0 +1,23 @@
+import React from "react"
+import { AppState } from "Router/state"
+import { Subscribe } from "unstated"
+
+type Mediator = {
+  trigger: (action: string, config?: object) => void
+}
+
+export const WithMediator: React.SFC<{
+  children: (mediator: Mediator) => React.ReactNode
+}> = ({ children }) => (
+  <Subscribe to={[AppState]}>
+    {({ state: { mediator } }) =>
+      children(
+        mediator || {
+          trigger() {
+            console.error("mediator is not present in AppState")
```

Comment:
👍 Nice. A good habit that has helped us time and again across the Reaction / Force / Sentry divide is to log errors with a little file-location metadata, like `console.error("Router/WithMediator mediator is not present in AppState")`. Not strictly necessary here but just a future thought.

Reply:
Yeah our source-map story is still broken in force :( though we do have uploads working -- what's missing is the associated sourcemap-to-release id, as well as needing to replace our very old in-house [fingerprinting lib](https://www.npmjs.com/package/bucket-assets) with webpack. Sourcemaps are 404 in prod since webpack isn't aware of bucket-assets rewriting filenames.)